src/node_snapshotable.cc

```diff
@@ -130,16 +140,48 @@ void SnapshotBuilder::Generate(SnapshotData* out,
                             nullptr,
                             node::EnvironmentFlags::kDefaultFlags,
                             {});
+
       // Run scripts in lib/internal/bootstrap/
       {
         TryCatch bootstrapCatch(isolate);
-        v8::MaybeLocal<Value> result = env->RunBootstrapping();
+        MaybeLocal<Value> result = env->RunBootstrapping();
         if (bootstrapCatch.HasCaught()) {
           PrintCaughtException(isolate, context, bootstrapCatch);
         }
         result.ToLocalChecked();
       }
 
+      // If --build-snapshot is true, lib/internal/main/mksnapshot.js would be
+      // loaded via LoadEnvironment() to execute process.argv[1] as the entry
+      // point (we currently only support this kind of entry point, but we
+      // could also explore snapshotting other kinds of execution modes
+      // in the future).
+      if (per_process::cli_options->build_snapshot) {
+#if HAVE_INSPECTOR
+        env->InitializeInspector({});
+#endif
+
+        TryCatch bootstrapCatch(isolate);
+        // TODO(joyeecheung): we could use the result for something special,
+        // like setting up initializers that should be invoked at snapshot
+        // dehydration.
+        MaybeLocal<Value> result =
+            LoadEnvironment(env, StartExecutionCallback{});
+        if (bootstrapCatch.HasCaught()) {
+          PrintCaughtException(isolate, context, bootstrapCatch);
+        }
+        result.ToLocalChecked();
+        // FIXME(joyeecheung): right now running the loop in the snapshot
+        // builder seems to introduces inconsistencies in JS land that need to
+        // be synchronized again after snapshot restoration.
+        int exit_code = SpinEventLoop(env).FromMaybe(1);
+        CHECK_EQ(exit_code, 0);
```

Comment:
```suggestion
        DCHECK_EQ(exit_code, 0);
```
If this can never happen, shouldn't we turn this into a debug check?

Reply:
As said above, I prefer to simply crash for now - it also makes debugging easier (no need to re-run to find out where the error is coming from, in case the bug doesn't reproduce easily) while the feature is still experimental.